corgi/tasks/pulp.py

```diff
@@ -0,0 +1,63 @@
+import logging
+
+from celery_singleton import Singleton
+
+from config.celery import app
+from corgi.collectors.pulp import Pulp
+from corgi.core.models import (
+    Channel,
+    ProductComponentRelation,
+    ProductStream,
+    ProductVariant,
+)
+from corgi.tasks.brew import fetch_modular_builds
+from corgi.tasks.common import RETRY_KWARGS, RETRYABLE_ERRORS, _create_relations
+
+logger = logging.getLogger(__name__)
+
+
+@app.task(base=Singleton, autoretry_for=RETRYABLE_ERRORS, retry_kwargs=RETRY_KWARGS)
+def setup_pulp_relations() -> None:
+    logger.info("Setting up CDN repo relations for all Channels")
+    for channel in Channel.objects.filter(type=Channel.Type.CDN_REPO):
+        for pv_ofuri in channel.product_variants:
+            pv = ProductVariant.objects.get(ofuri=pv_ofuri)
+            slow_setup_pulp_rpm_relations.delay(channel.name, pv.name)
+            slow_setup_pulp_module_relations.delay(channel.name, pv.name)
+
+
+@app.task(base=Singleton, autoretry_for=RETRYABLE_ERRORS, retry_kwargs=RETRY_KWARGS)
+def slow_setup_pulp_rpm_relations(channel, variant):
+    srpm_build_ids = Pulp().get_rpm_data(channel)
+    no_of_relations = _create_relations(
+        srpm_build_ids, channel, variant, ProductComponentRelation.Type.CDN_REPO
+    )
+    if no_of_relations > 0:
+        logger.info("Created %s new relations for SRPMs in %s", no_of_relations, channel)
+
+
+@app.task(base=Singleton, autoretry_for=RETRYABLE_ERRORS, retry_kwargs=RETRY_KWARGS)
+def slow_setup_pulp_module_relations(channel, variant):
+    module_build_ids = Pulp().get_module_data(channel)
+    no_of_relations = _create_relations(
+        module_build_ids, channel, variant, ProductComponentRelation.Type.CDN_REPO
+    )
+    if no_of_relations > 0:
+        logger.info("Created %s new relations for rhel_modules in %s", no_of_relations, channel)
+
+
+def get_builds_by_cdn_repo(stream_names: list[str]):
```

Comment:
It looks like this is only called when the management command is run manually. If that's the case, it might make more sense to move this into the loadpulpdata.py module. Or if it really is a Celery task, we should add the app.task() decorator here as well.

Reply:
OK, will do.